numpy/_core/src/umath/override.c

```diff
@@ -148,18 +148,16 @@ static int
 normalize_signature_keyword(PyObject *normal_kwds)
 {
     /* If the keywords include `sig` rename to `signature`. */
-    PyObject* obj = _PyDict_GetItemStringWithError(normal_kwds, "sig");
-    if (obj == NULL && PyErr_Occurred()) {
+    PyObject* obj = NULL;
+    int result = PyDict_GetItemStringRef(normal_kwds, "sig", &obj);
+    if (result == -1) {
         return -1;
     }
-    if (obj != NULL) {
-        /*
-         * No INCREF or DECREF needed: got a borrowed reference above,
-         * and, unlike e.g. PyList_SetItem, PyDict_SetItem INCREF's it.
-         */
+    if (result == 1) {
         if (PyDict_SetItemString(normal_kwds, "signature", obj) < 0) {
```

Comment:
missing: Py_DECREF(obj)

Reply:
besides the DECREF two lines below?

Reply:
PyDict_GetItemStringRef(&obj) creates a strong reference. This line (`return -1;`) does exit the function. I don't see anything which clears the strong reference.